### DISCUSSION TITLE
apollo-v9.0.0

### DISCUSSION THREAD

**Discussion by diegohu** [ApolloAuto/apollo#15231] [General]
Apollo Open Source Platform 9.0 further focuses on enhancing the development and debugging experience, dedicated to providing autonomous driving developers with a unified development tool platform and easy-to-extend PnC and perception software framework interfaces. The new version reshapes the PnC and perception extension development method based on package management. It optimizes component splitting and configuration management according to business logic, simplifying the process of calling. In addition to the component extension method, a more lightweight plugin extension method has been added, simplifying the process of extending. The new version introduces Dreamview Plus,  a brand-new developer tool that introduces modes for convenient multi-scenario use, a panel layout for customizing visualization, and a resource center providing richer development resources. Furthermore, the LiDAR and Camera detection models in the new version have been upgraded for improved results, and incremental training methods have been opened up for easy extension. At the same time, support for 4D millimeter-wave radar has been added. Finally, the new version is adapted to the ARM architecture, and supports compilation and running on Orin, providing developers with additional device options.

## Major New Features and Improvements:
* PnC Extension Development Pattern Based On Package Management
  * Unified external interfaces to decouple the operation layer and PNC module.
  * A brand-new plugin extension method to facilitate developers in developing and deploying their functionality. 
  * Global parameters and local parameters are divided to allow developers to query and modify parameters.

* Perception Extension Development Pattern Based On Package Management
  * Re-split the perception components based on  "functional" granularity to facilitate reuse.
  * A brand-new plugin development mode to facilitate the replacement of algorithms under the existing perception pipeline.
  * Simplified and unified configuration to allow developers to query and modify parameters at any time.

* Brand New Dreamview Plus Developer Tool
  * Organize the usage scenarios of the development tool based on "mode", such as Perception mode, PnC mode, and Vehicle Test mode.
  * Encapsulate each visualization functionality into an independent panel and support developers to customize the panel.
  * Provide various resources such as maps, scenarios, vehicle configurations, and data records in the resource center to facilitate development and debugging.

* Fully Upgraded Perception Model to Support Incremental Training
  * CenterPoint is adopted to replace CNNSeg model by default in the LiDAR perception pipeline and YOLOX+YOLO3D is adopted to replace the original YOLO model by default in the Camera perception pipeline.
  * Provide incremental training to improve perception model capabilities in specific scenarios by using a small amount of annotated data and Apollo pre-trained models.
  * Support 4D millimeter-wave radar from hardware driver to perception model layer.

**[Note]** All models and methodologies included in Apollo Open Source Platform 9.0 are for research purposes only. Productized and commercialized uses of these models are **NOT** encouraged, and it is at your own risk. Please be cautious to try Apollo Open Source Platform 9.0 with sufficient safety protection mechanisms in place. Your feedback is highly appreciated so that we can continuously improve the models.

<hr /><em>This discussion was created from the release <a href='https://github.com/ApolloAuto/apollo/releases/tag/v9.0.0'>apollo-v9.0.0</a>.</em>

**Comment by howicc** [comment]
Great work. Do we have support for NVidia 40xx series in this version?

**Reply by daohu527** [reply]
We already support NVidia 40xx series. You can refer to https://github.com/ApolloAuto/apollo/discussions/15219#discussioncomment-7772518

**Comment by confuse05** [comment]
I came across the introduction article on version 9.0, where it mentioned that the PNC capability provides safety mechanisms such as anomaly detection and fault alerts. I would like to know where I can find detailed instructions on accessing fault displays, configuring monitoring content, and other related functionalities. Could you please guide me on this?

**Reply by diegohu** [reply]
Regarding the functional safety mechanism you mentioned, it is a capability provided by Apollo Open Platform 9.0 for enterprise customers in closed-venue low-speed scenarios. If you are interested in this, you can contact us via email at [apollox@baidu.com](mailto:apollox@baidu.com) or scan the QR code to fill in your cooperation needs.
![95188a9675a61304d1ce259bb550d04d64d2bd48](https://github.com/ApolloAuto/apollo/assets/6268966/cee1cd31-74f2-4f41-a58e-b85ab876dd1e)

**Comment by Ariel-0228** [comment]
How to compile and run Apollo 9.0 on Orin， I'm having a lot of problems using the method guide document steps, I wonder if there are any mature guide steps?
